### ISSUE TITLE
code

### ISSUE THREAD

**Issue by ABHINAB-CHHETRI** [ABHINAB-CHHETRI/ChessEngine_in_CPP_from_Scratch#1]
import numpy as np
from sklearn.datasets import load_iris

# Load and preprocess the Iris dataset
def load_and_preprocess_iris():
    iris = load_iris()
    dataset = np.c_[iris.data, iris.target]  # Combine features and labels
    np.random.shuffle(dataset)  # Shuffle the dataset
    train_data = dataset[:120, :-1]  # Training features
    train_labels = dataset[:120, -1].astype(int)  # Training labels
    test_data = dataset[120:, :-1]  # Testing features
    test_labels = dataset[120:, -1].astype(int)  # Testing labels
    return train_data, train_labels, test_data, test_labels

# Activation functions
def relu(x):
    return np.maximum(0, x)

def softmax(x):
    exp_x = np.exp(x)
    row_sums = np.sum(exp_x, axis=1, keepdims=True)
    return exp_x / row_sums

# NeuronLayer class
class NeuronLayer:
    def __init__(self, num_inputs, num_neurons):
        # Initialize weights and biases
        self.weights = np.random.rand(num_neurons, num_inputs) * 0.01
        self.biases = np.random.rand(num_neurons)

    def forward(self, inputs):
        # Forward pass: matrix multiplication + bias
        return np.dot(inputs, self.weights.T) + self.biases

# Building and using the neural network
train_data, train_labels, test_data, test_labels = load_and_preprocess_iris()

# Create layers
layer1 = NeuronLayer(num_inputs=4, num_neurons=8)  # First hidden layer (4 inputs, 8 neurons)
layer2 = NeuronLayer(num_inputs=8, num_neurons=3)  # Output layer (8 inputs, 3 neurons)

# Forward pass through layers
layer1_output = relu(layer1.forward(train_data))  # Forward pass through first layer with ReLU
layer2_output = softmax(layer2.forward(layer1_output))  # Forward pass through second layer with Softmax

# Display predictions for the first 5 training samples
print("Predicted probabilities for first 5 samples:")
print(layer2_output[:5])
